### PR TITLE
rf-245 Combined self user into userInfos in AvatarCanvas

### LINKED ISSUE #245: [Refactor]: Add selfUserInfo to UserInfoContext
### Refactor Request

Since there are multiple places that uses `selfUserInfo`, it might be better if `selfUserInfo` live in `UserInfoContext` to avoid passing around in props everywhere

#### Uses
- AvatarCanvas
- Sidebar

### PR DESCRIPTION
**Pull request by Njima1572**
## Which Issue was this PR made for?

- Closes #245 

## What is within the scope with this PR

- Added Self user info into userinfos

## Other sidenotes that reviewer should be aware of

- I dont like this implementation with id==="self" thing

### CODE REVIEW

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

File: client/src/components/AvatarCanvas/index.tsx [hunk 1]
```diff
@@ -37,41 +39,26 @@ function AvatarCanvas({
...
-                initial={info.name[0]}
-                active={info.active}
-                mute={info.mute}
-              />
-            </Draggable>
-          );
-        })
-      }
+      {userInfos.map((info, index) => {
+        if (!info) {
```
Comment (siberowl) [inline client/src/components/AvatarCanvas/index.tsx:43]: We can make a follow up ticket for this, but probably should just delete the user if the info doesn't exist, instead of creating a grey (or whatever the default was), empty avatar.

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

File: client/src/pages/RoomPage/index.tsx [hunk 2]
```diff
@@ -30,6 +29,7 @@ function RoomPage({ name }: { name: string }): JSX.Element {
   const [selfUserInfo, setSelfUserInfo] = useState<UserInfo>({ ...defaultUserInfo, name });
```
Comment (siberowl) [inline client/src/pages/RoomPage/index.tsx:29]: Since self user info still exists, might want to edit the title to be more clear?

File: client/src/pages/RoomPage/index.tsx [hunk 3]
```diff
@@ -191,8 +192,15 @@ function RoomPage({ name }: { name: string }): JSX.Element {
   }, []);
 
   useEffect(() => {
-    globalUserInfos = userInfos;
-  }, [userInfos]);
+    if (stream) {
```
Comment (siberowl) [inline client/src/pages/RoomPage/index.tsx:195]: Seems unrelated? A bit confused about this

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

File: client/src/pages/RoomPage/index.tsx (hunk 2 above)
Comment (Njima1572) [reply client/src/pages/RoomPage/index.tsx:29]: Changed title

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

File: client/src/pages/RoomPage/index.tsx (hunk 3 above)
Comment (Njima1572) [reply client/src/pages/RoomPage/index.tsx]: Seems like a residue from when I was merging main into this

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-296.dev.mochatto.com

File: client/src/components/AvatarCanvas/index.tsx (hunk 1 above)
Comment (Njima1572) [reply client/src/components/AvatarCanvas/index.tsx]: Updated with https://github.com/Amijakan/mochatto/pull/296/commits/631c08f31d3fa785e6f3b2089b43e894ec46b923

**Review by siberowl — APPROVED** [review]
